x-pack/plugins/security_solution/server/lib/detection_engine/rule_management/utils/validate.ts

```diff
@@ -51,3 +65,45 @@ export const transformValidateBulkError = (
     });
   }
 };
+
+// Temporary functionality until the new System Actions is in place
+// for now we want to make sure that user cannot configure Isolate action if has no RBAC permissions to do so
+export const validateResponseActionsPermissions = async (
+  securitySolution: SecuritySolutionApiRequestHandlerContext,
+  ruleUpdate: RuleCreateProps | RuleUpdateProps,
+  existingRule?: RuleAlertType | null
+) => {
+  const { experimentalFeatures } = await securitySolution.getConfig();
+
+  if (!experimentalFeatures.endpointResponseActionsEnabled || !isQueryRule(ruleUpdate.type)) {
+    return;
+  }
+
+  const payload = ruleUpdate as QueryRule;
+  const existingPayload = existingRule as Rule<UnifiedQueryRuleParams>;
+
+  if (payload.response_actions?.length || existingPayload?.params?.responseActions?.length) {
+    const endpointAuthz = await securitySolution.getEndpointAuthz();
+
+    const difference = findDifferenceInArrays<ResponseAction, RuleResponseAction>(
+      payload.response_actions,
+      existingPayload?.params?.responseActions
+    );
+
+    difference.forEach((action) => {
+      if ('command' in action?.params) {
```

Comment:
I don't see the API restricting this to only `isolate`. Did you forget to make that check here?

I know you are trying to build it in a way that it will just work with other aciton as they are introduced, but the API needs to be locked down to only the actions we support currently.